sormas-base/setup/s2s-generate-cert.sh

```diff
@@ -53,73 +53,90 @@ else
   else
-    while [[ ! -f "${SORMAS_PROPERTIES}" ]]; do
-		  read -r -p "Please specify a valid sormas properties path: " SORMAS_PROPERTIES
+     while [[ ! -d "${SORMAS_DOMAIN_DIR}" ]]; do
+		  read -r -p "Please specify a valid sormas domain path: " SORMAS_DOMAIN_DIR
 	  done
-	  export SORMAS_PROPERTIES
   fi
+fi
+
+SORMAS_PROPERTIES="${SORMAS_DOMAIN_DIR}/sormas.properties"
+
+while [[ -z "${SORMAS_ORG_ID}" ]]; do
+  read -p "Please provide an ID for the organization: " SORMAS_ORG_ID
+done
+
+while [[ -z "${SORMAS_ORG_NAME}" ]]; do
+  read -p "Please provide the Name of your organization: " SORMAS_ORG_NAME
+done
+
+while [[ -z "${SORMAS_HOST_NAME}" ]]; do
+  read -p "Please provide the Hostname of the server: " SORMAS_HOST_NAME
+done
+
+read -p "Please provide the https port of the server (443): " SORMAS_HTTPS_PORT
+if [[ -z "${SORMAS_HTTPS_PORT}" ]]; then
+  SORMAS_HOST_AND_PORT="${SORMAS_HOST_NAME}";
 else
-  if [[ ! -f "${SORMAS_PROPERTIES}" ]]; then
-    echo "sormas properties file not found: ${SORMAS_PROPERTIES}"
-    exit 1
-  fi
+  SORMAS_HOST_AND_PORT="${SORMAS_HOST_NAME}:${SORMAS_HTTPS_PORT}";
 fi
 
 while [[ -z "${SORMAS_S2S_CERT_PASS}" ]] || [[ ${#SORMAS_S2S_CERT_PASS} -lt 6 ]]; do
   read -sp "Please provide a password for the certificate (at least 6 characters): " SORMAS_S2S_CERT_PASS
   echo
 done
 
-while [[ -z "${SORMAS_S2S_CERT_CN}" ]]; do
-  read -p "Please provide a Common Name (CN) for the certificate: " SORMAS_S2S_CERT_CN
-done
-
-while [[ -z "${SORMAS_S2S_CERT_ORG}" ]]; do
-  read -p "Please provide an Organization (O) for the certificate: " SORMAS_S2S_CERT_ORG
+while [[ -z "${SORMAS_S2S_REST_PASSWORD}" ]] || [[ ${#SORMAS_S2S_REST_PASSWORD} -lt 12 ]]; do
+  read -sp "Please provide a password for the REST interface (at least 12 characters): " SORMAS_S2S_REST_PASSWORD
+  echo
 done
 
 if [[ ${LINUX} = true ]]; then
-  CERT_SUBJ="/CN=${SORMAS_S2S_CERT_CN}/OU=SORMAS/O=${SORMAS_S2S_CERT_ORG}"
+  CERT_SUBJ="/CN=${SORMAS_ORG_ID}/OU=SORMAS/O=${SORMAS_ORG_NAME}"
 else
-  CERT_SUBJ="//CN=${SORMAS_S2S_CERT_CN}\OU=SORMAS\O=${SORMAS_S2S_CERT_ORG}"
+  CERT_SUBJ="//CN=${SORMAS_ORG_ID}\OU=SORMAS\O=${SORMAS_ORG_NAME}"
 fi
 echo "The certificate will be generated with the following subject:"
-echo "CN=${SORMAS_S2S_CERT_CN},OU=SORMAS,O=${SORMAS_S2S_CERT_ORG}"
+echo "CN=${SORMAS_ORG_ID},OU=SORMAS,O=${SORMAS_ORG_NAME}"
 read -p "Press [Enter] to continue or [Ctrl+C] to cancel."
 
-PEM_FILE=${SORMAS2SORMAS_DIR}/sormas2sormas.privkey.pem
-P12_FILE=${SORMAS2SORMAS_DIR}/sormas2sormas.keystore.p12
-CRT_FILE=${SORMAS2SORMAS_DIR}/sormas2sormas.cert.crt
-CSV_FILE=${SORMAS2SORMAS_DIR}/server-access-data.csv
+PEM_FILE=${SORMAS2SORMAS_DIR}/${SORMAS_HOST_NAME}.sormas2sormas.privkey.pem
+P12_FILE_NAME=${SORMAS_HOST_NAME}.sormas2sormas.keystore.p12
+P12_FILE=${SORMAS2SORMAS_DIR}/${P12_FILE_NAME}
+CRT_FILE=${SORMAS2SORMAS_DIR}/${SORMAS_HOST_NAME}.sormas2sormas.cert.crt
+CSV_FILE_NAME=${SORMAS_HOST_NAME//./-}-server-access-data.csv;
```

Comment:
why is (only) this file name normalized this way?

Reply:
because the constant part written with dashes, I just wanted to keep it consistent and I'm not sure if windows can handle files with multiple dots in the name when someone tries to open it.

I'm not a windows user since 10 years ago but I remember some issues caused by the naming of files